src/elements.js

```diff
@@ -488,8 +488,14 @@ _.register({
 			"time": /^[H\d]{2}:[M\d]{2}/i,
 			"datetime-local": /^[Y\d]{4}-[M\d]{2}-[D\d]{2} [H\d]{2}:[Mi\d]{2}/i,
 			"date": /^[Y\d]{4}-[M\d]{2}-[D\d]{2}$/i,
+			"duration": /^P/,
+        	"duration-normal": /(day[s]?|hour[s]?|minute[s]?|second[s]?)$/,
+        	"duration-abbreviation": /(d|h|m|s)$/,
 		},
 		defaultFormats: {
+			"duration": name => "[getduration(".concat(name,"\"officialDurationString\")]"),
+        	"duration-abbreviation": name => "[getduration(".concat(name,"\"abbreviatedDurationString\")]"),
+        	"duration-normal": name => "[getduration(".concat(name,"\"normalDurationString\")]"),
```

Comment:
Please use template strings for concatenation, which are more readable than `.concat()` method calls, and don't require escaping of double quotes. See the other values in this object literal for example.